cdap-ui/app/cdap/components/AbstractWidget/InputFieldDropdown/index.tsx

```diff
@@ -52,14 +62,53 @@ const InputFieldDropdown: React.FC<IInputFieldProps> = ({
   onChange,
   disabled,
   extraConfig,
+  widgetProps,
 }) => {
   const inputSchema = objectQuery(extraConfig, 'inputSchema');
-  const fieldValues = getFields(inputSchema);
-  const widgetProps = {
-    options: fieldValues,
-  };
 
-  return <Select value={value} onChange={onChange} widgetProps={widgetProps} disabled={disabled} />;
+  const isMultiSelect: boolean = widgetProps.multiselect || false;
+  const allowedTypes: string[] = widgetProps.allowedTypes || [];
+
+  const fieldValues = getFields(inputSchema, allowedTypes);
+
+  const newValue = value
+    .toString()
+    .split(delimiter)
+    .filter((v) => fieldValues.includes(v))
+    .toString();
```

Comment:
the last `toString()` should be `join(delimiter)` , right?

Reply:
Yeah, fixed it!